Add min-height to scc container to avoid cutting off subject heading results

danamansana opened the pull request:
**What's this do?**

- Adds a configurable class to the content of the SCC container
- For subjects page, configures this class to indicate the subjects page
- Adds styling to give primary content on subjects page a min height
- Adds logic to adjust the min-height if necessary.

**Why are we doing this? (w/ JIRA link if applicable)**
- We want to stop the subject search box results from being cut off when there are no results for the current search

**How should this be QAed?**
- On the main page, select subject search and enter some nonsense
- You should see the subject index page with a message that there are no results. Now start typing into the search box. It should not be cut off

**Dependencies for merging? Releasing to production?**
No

**Has the application documentation been updated for these changes?**
No

**Did someone actually run this code to verify it works?**
Yes.

nonword commented on src/app/components/SubjectHeading/Search/SubjectHeadingSearch.jsx:
```diff
@@ -35,6 +35,8 @@ class SubjectHeadingSearch extends React.Component {
         return axios(`${appConfig.baseUrl}/api/subjectHeadings/autosuggest?query=${this.state.userInput}`)
           .then((res) => {
+            const numberOfResults = res.data.autosuggest.length;
+            document.getElementById('subject-heading-content-primary').style['min-height'] = `${(numberOfResults > 12 ? numberOfResults : 12) * 50}px`
```
Is `document.getElementById` discouraged in React or nah? Is it possible `document.getElementById('subject-heading-content-primary')` is going to match 0 nodes and this will error?

nonword reviewed (commented):
This version creates a <style> block in the main template just so that we can write page specific CSS into it as people type in the subject search. I kind of think directly modifying the container style was better because it was more direct. Long term, I think a `ref` is what's needed here, but I don't know how to use React v16 `ref`s. In the short term, I think using `document.getElementById` and directly modifying the container min-height as a function of numResults is good. Just be wary about the id selector failing to find the element due to a React repaint. Did you find `document.getElementById` failing to find the container and that's why you switched to this inline style approach?

danamansana commented:
Sorry, didn't mean for you to review the previous version with the <style> block, I decided that is not a good approach.
I've changed the PR to follow the following strategy: 

1. the SCC container accepts a param which sets the id of the `content-primary` element, the `SubjectHeadingsIndexPage` sets that param to a default value, and we style using css. 
2. In case that default value doesn't work, the SCC container also accepts an inline style for the `content-primary` element, which the `SubjectHeadingsIndexPage` also sets, but in a programmatic way using React hooks. It passes the setter for that hook to the `SubjectHeadingSearch` component, which can then change the style if necessary.

This addresses your concerns raised above by staying completely within the React API, and setting appropriate default values for everything.
